sklearn/linear_model/_coordinate_descent.py

```diff
@@ -146,47 +149,35 @@ def _alpha_grid(
             "your estimator with the appropriate `alphas=` "
             "argument."
         )
-    n_samples = len(y)
-
-    sparse_center = False
-    if Xy is None:
-        X_sparse = sparse.isspmatrix(X)
-        sparse_center = X_sparse and (fit_intercept or normalize)
-        X = check_array(
-            X, accept_sparse="csc", copy=(copy_X and fit_intercept and not X_sparse)
+    if Xy is not None and ((sample_weight is None) or (fit_intercept is False)):
```

Comment:
Maybe we should warn if `Xy` is not None but is never used.
WDYT?

Reply:
Fixed so that Xy is always used when it is not None.

Reply:
My old code was probably correct. My memory just isn't so good, so I forgot why exactly I coded it that way.
I'm not even sure if it's even possible for `Xy` *not* to be `None`.
It's honestly kind of hard to read the linear_model source code and figure out what's going on.